### PR TITLE
{Container app} `az containerapp env certificate upload`: Fix bumping `pyOpenSSL` to `24.0.0`, `OpenSSL.crypto.loads_pkcs12` is removed in `23.3.0`

### PR DESCRIPTION
**Pull request by Greedygre**
**Related command**

**Description**

At PR: https://github.com/Azure/azure-cli/pull/28306/files
azure-cli will update version from `pyOpenSSL==23.2.0` to `pyOpenSSL==24.0.0`
OpenSSL.crypto.loads_pkcs12 is removed in `23.3.0`.
Ref: https://www.pyopenssl.org/en/latest/changelog.html

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Containberapp

**Review by zhoxing-ms — DISMISSED** [review]
Please also get review and approve from your team, thank you~

**Comment by zhoxing-ms** [comment]
@bebound Please help review this PR as well~

File: src/azure-cli/azure/cli/command_modules/containerapp/_utils.py [hunk 1]
```diff
@@ -1477,12 +1479,13 @@ def load_cert_file(file_path, cert_password=None):
             elif os.path.splitext(file_path)[1] in ['.pfx']:
                 cert_data = f.read()
                 try:
-                    p12 = crypto.load_pkcs12(cert_data, cert_password)
+                    # The password to use to decrypt the data. None if the PKCS12 is not encrypted.
+                    cert_password_bytes = cert_password.encode('utf-8') if cert_password is not None else None
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/_utils.py:1483]: what if the cert_password is ""

File: src/azure-cli/azure/cli/command_modules/containerapp/_utils.py [hunk 2]
```diff
@@ -1477,12 +1479,13 @@ def load_cert_file(file_path, cert_password=None):
...
+                    p12 = pkcs12.load_pkcs12(cert_data, cert_password_bytes)
                 except Exception as e:
                     raise FileOperationError('Failed to load the certificate file. This may be due to an incorrect or missing password. Please double check and try again.\nError: {}'.format(e)) from e
-                x509 = p12.get_certificate()
-                digest_algorithm = 'sha256'
-                thumbprint = x509.digest(digest_algorithm).decode("utf-8").replace(':', '')
+                x509 = p12.cert.certificate
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/containerapp/_utils.py:1487]: add null check?

File: src/azure-cli/azure/cli/command_modules/containerapp/_utils.py (hunk 1 above)
Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_utils.py:1483]: Add test for password is "" in `test_load_cert_file`.

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_utils.py:1483]: If Password is "", it means the PKCS12 is not encrypted.

File: src/azure-cli/azure/cli/command_modules/containerapp/_utils.py [hunk 3]
```diff
@@ -1477,12 +1479,15 @@ def load_cert_file(file_path, cert_password=None):
...
-                    p12 = crypto.load_pkcs12(cert_data, cert_password)
+                    # The password to use to decrypt the data. None if the PKCS12 is not encrypted.
+                    cert_password_bytes = cert_password.encode('utf-8') if cert_password else None
+                    p12 = pkcs12.load_pkcs12(cert_data, cert_password_bytes)
                 except Exception as e:
                     raise FileOperationError('Failed to load the certificate file. This may be due to an incorrect or missing password. Please double check and try again.\nError: {}'.format(e)) from e
-                x509 = p12.get_certificate()
-                digest_algorithm = 'sha256'
-                thumbprint = x509.digest(digest_algorithm).decode("utf-8").replace(':', '')
+                if p12 is None or p12.cert is None or p12.cert.certificate is None:
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/containerapp/_utils.py:1487]: According to the doc, `p12.cert.certificate` is `Certificate` and can't be `None`.
https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#cryptography.hazmat.primitives.serialization.pkcs12.PKCS12Certificate

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_utils.py]: Fixed, please help to review again. Thanks!

Comment (Greedygre) [reply src/azure-cli/azure/cli/command_modules/containerapp/_utils.py]: Hi @bebound  Fixed again, thanks.

**Review by Juliehzl — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @bebound 

Can you help to review and approve this PR? Thanks!

**Review by bebound — APPROVED** [review]

**Pull request merged**